src/Symfony/Component/Serializer/Tests/Normalizer/DateTimeNormalizerTest.php

```diff
@@ -235,12 +235,24 @@ public function testSupportsDenormalization()
         $this->assertFalse($this->normalizer->supportsDenormalization('foo', 'Bar'));
     }
 
+    public function testSupportsDenormalizationUsingSupportedTypesPassedInConstructor()
+    {
+        $normalizer = new DateTimeNormalizer(supportedTypes: [DateTimeChild::class => true, DateTimeImmutableChild::class => true]);
```

Comment:
I would like to keep the constructor simple and really only pass an array of supported types instead of that internal structure of an array keyed by the type.

```suggestion
        $normalizer = new DateTimeNormalizer(supportedTypes: [DateTimeChild::class, DateTimeImmutableChild::class]);
```